### PR TITLE
Use Long to represent Content-Length

### PR DESCRIPTION
**Pull request by mikkokar**

### CODE REVIEW

File: components/api/src/main/java/com/hotels/styx/api/HttpResponse.java [hunk 1]
```diff
@@ -571,7 +571,7 @@ Builder ensureContentLengthIsValid() {
 
         private static boolean isInteger(String contentLength) {
             try {
-                parseInt(contentLength);
+                parseLong(contentLength);
```
Comment (dvlato) [inline components/api/src/main/java/com/hotels/styx/api/HttpResponse.java]: do you think we should validate it is a positive integer / greater than 0?

File: components/api/src/main/java/com/hotels/styx/api/HttpRequest.java [hunk 2]
```diff
@@ -728,10 +728,10 @@ private void ensureContentLengthIsValid() {
             return headerValues.isEmpty() ? Optional.empty() : Optional.of(headerValues.get(0));
         }
 
-        private static boolean isInteger(String contentLength) {
+        private static boolean isPositiveInteger(String contentLength) {
```
Comment (kvosper) [inline components/api/src/main/java/com/hotels/styx/api/HttpRequest.java]:
```suggestion
        private static boolean isNonNegativeInteger(String contentLength) {
```
Generally the term "positive integer" refers to an integer greater than zero. I'm not sure if there is an official ruling for numbers in Java, but this is the use of the term I am familiar with.

File: components/api/src/main/java/com/hotels/styx/api/LiveHttpRequest.java [hunk 3]
```diff
@@ -1017,7 +1017,7 @@ private void ensureContentLengthIsValid() {
 
         private static boolean isInteger(String contentLength) {
```
Comment (kvosper) [inline components/api/src/main/java/com/hotels/styx/api/LiveHttpRequest.java]: So live requests + responses do not have the `>= 0` check that full ones have?

File: components/api/src/main/java/com/hotels/styx/api/HttpResponse.java (hunk 1 above)
Comment (mikkokar) [reply components/api/src/main/java/com/hotels/styx/api/HttpResponse.java]: Done

File: components/api/src/main/java/com/hotels/styx/api/LiveHttpRequest.java (hunk 3 above)
Comment (mikkokar) [reply components/api/src/main/java/com/hotels/styx/api/LiveHttpRequest.java]: My mistake. Got distracted when implementing this. :-/

Added.

**Pull request merged**